src/plugin/algoSignMessage.ts

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Interfaces, Models, Errors } from '@open-rights-exchange/chain-js'
+import {
+  AlgoSignDataInput,
+  AlgorandPrivateKey,
+} from './models'
+import { sign } from "./algoCrypto"
+
+export class AlgorandSignMessage implements Interfaces.SignMessage {
+  constructor(data: any, options?: Models.SignMessageOptions ) {
+    this.applyOptions(options)
+    this.applyData(data)
+    this.setSignMethod()
+    this._isValidated = false
+  }
+
+  private _isValidated: boolean
+
+  private _signMethod: string
+
+  private _options: Models.SignMessageOptions
+
+  private _data: any
+
+  private applyOptions(options: Models.SignMessageOptions) {
+    this._options = options ? options : { signMethod: 'algorand-sign'}
```

Comment:
This value should be 'default' so that it is the same across all chains:
e.g. SignMethod.Default - 'default'
Use an Enum and export the type like other types